python/ray/autoscaler/_private/aliyun/utils.py

```diff
@@ -0,0 +1,344 @@
+import logging
+import json
+
+from aliyunsdkcore import client
+from aliyunsdkcore.acs_exception.exceptions \
+    import ClientException, ServerException
+from aliyunsdkecs.request.v20140526.CreateInstanceRequest \
+    import CreateInstanceRequest
+from aliyunsdkecs.request.v20140526.DeleteInstanceRequest \
+    import DeleteInstanceRequest
+from aliyunsdkecs.request.v20140526.DeleteInstancesRequest \
+    import DeleteInstancesRequest
+from aliyunsdkecs.request.v20140526.StartInstanceRequest \
+    import StartInstanceRequest
+from aliyunsdkecs.request.v20140526.StopInstanceRequest \
+    import StopInstanceRequest
+from aliyunsdkecs.request.v20140526.StopInstancesRequest \
+    import StopInstancesRequest
+from aliyunsdkecs.request.v20140526.DescribeInstancesRequest \
+    import DescribeInstancesRequest
+from aliyunsdkecs.request.v20140526.TagResourcesRequest \
+    import TagResourcesRequest
+from aliyunsdkecs.request.v20140526.AllocatePublicIpAddressRequest \
+    import AllocatePublicIpAddressRequest
+from aliyunsdkecs.request.v20140526.AttachKeyPairRequest \
+    import AttachKeyPairRequest
+from aliyunsdkecs.request.v20140526.ImportKeyPairRequest \
+    import ImportKeyPairRequest
+from aliyunsdkecs.request.v20140526.DescribeKeyPairsRequest \
+    import DescribeKeyPairsRequest
+from aliyunsdkecs.request.v20140526.CreateKeyPairRequest \
+    import CreateKeyPairRequest
+from aliyunsdkecs.request.v20140526.RunInstancesRequest \
+    import RunInstancesRequest
+from aliyunsdkecs.request.v20140526.CreateSecurityGroupRequest \
+    import CreateSecurityGroupRequest
+from aliyunsdkecs.request.v20140526.DeleteKeyPairsRequest \
+    import DeleteKeyPairsRequest
+from aliyunsdkecs.request.v20140526.AuthorizeSecurityGroupRequest \
+    import AuthorizeSecurityGroupRequest
+from aliyunsdkecs.request.v20140526.DescribeSecurityGroupsRequest \
+    import DescribeSecurityGroupsRequest
+from aliyunsdkecs.request.v20140526.CreateVSwitchRequest \
+    import CreateVSwitchRequest
+from aliyunsdkecs.request.v20140526.CreateVpcRequest \
+    import CreateVpcRequest
+from aliyunsdkecs.request.v20140526.DescribeZonesRequest \
+    import DescribeZonesRequest
+from aliyunsdkecs.request.v20140526.DescribeVpcsRequest \
+    import DescribeVpcsRequest
+from aliyunsdkecs.request.v20140526.DescribeVSwitchesRequest \
+    import DescribeVSwitchesRequest
+from aliyunsdkecs.request.v20140526.RebootInstanceRequest \
+    import RebootInstanceRequest
+
+
+class AcsClient:
```

Comment:
Could you add a quick class comment for this class? It appears to be a layer of abstraction between the NodeProvider class and the actual implementation on Aliyun.

Reply:
Add comment for this class.